packages/core/src/indexing/database-index-api.ts

```diff
@@ -79,6 +80,14 @@ export class DatabaseIndexApi<DateType = Date | number> {
    * @param models
    */
   async indexModels(models: Array<IndexModelArgs>): Promise<void> {
+    for (const modelArgs of this.previouslyIndexedModels) {
+      const modelPreviouslyIndexed = this.previouslyIndexedModels.some(function (streamId) {
+        return String(streamId) === String(modelArgs)
+      })
+      if (modelPreviouslyIndexed){
+        throw new Error(`Cannot re-index model ${modelArgs.model.toString()}, data may not be up-to-date`)
+      }
+    }
     await this.indexModelsInDatabase(models)
     for (const modelArgs of models) {
```

Comment:
Previously updated models should be updated here?

Reply:
That seems consistent with a DX I would want. I'm wondering if this is out of scope of the ticket since that was previously the description and then crossed out in favor of disallowing re-indexing streams that have had their is_indexed field set to false. I'd like to understand the historical context behind striking through the previous description and will seek some clarity on that.

Reply:
Yes, thats a good DX behavior for now.